storage/replica.go

```diff
@@ -512,7 +526,7 @@ func (r *Replica) setDesc(desc *roachpb.RangeDescriptor) error {
 // setDescWithoutProcessUpdate updates the range descriptor without calling
 // processRangeDescriptorUpdate.
 func (r *Replica) setDescWithoutProcessUpdate(desc *roachpb.RangeDescriptor) {
```

Comment:
no need, added the lock within this function instead.  It never gets called form a locked function.